app/enterprise/0.33-x/developer-portal/best-practice.md

```diff
@@ -0,0 +1,204 @@
+│       │   ├── _header.scss
+│       │   ├── _headerDropdown.scss
+│       │   └── _sidebar.scss
+│       ├── pages
+│       │   ├── _404.scss
+│       │   ├── _guides.scss
+│       │   ├── _index.scss
+│       │   └── login.scss
+│       ├── theme-css.hbs
+│       └── util
+│           ├── _fixes.scss
+│           └── _spec-ui.scss
+├── index.html
+├── js
+│   ├── Api.js
+│   ├── Auth.js
+│   ├── HbsManager.js
+│   ├── config.js
+│   ├── dashboard
+│   │   ├── App.vue
+│   │   ├── components
+│   │   │   ├── ChartControls.vue
+│   │   │   ├── ChartTooltip.vue
+│   │   │   ├── CredentialForm.vue
+│   │   │   ├── CredentialFormModal.vue
+│   │   │   ├── CredentialTable.vue
+│   │   │   ├── DevRequestsChart.vue
+│   │   │   └── LineChart.vue
+│   │   ├── mixins
+│   │   │   ├── polling.js
+│   │   │   └── toaster.js
+│   │   ├── pages
+│   │   │   └── dashboard.vue
+│   │   └── schemas
+│   │       ├── ACL.js
+│   │       ├── BasicAuth.js
+│   │       ├── HMAC.js
+│   │       ├── JWT.js
+│   │       ├── KeyAuth.js
+│   │       ├── OAuth2.js
+│   │       └── plugins.js
+│   ├── dashboard.js
+│   ├── site.js
+│   └── utils
+│       ├── common.js
+│       ├── error.js
+│       ├── route.js
+│       ├── spec.js
+│       └── vitals.js
+└── templates
+    ├── pages
+    │   ├── 404.hbs
+    │   ├── about.hbs
+    │   ├── dashboard.hbs
+    │   ├── documentation
+    │   │   ├── api1.hbs
+    │   │   ├── api2.hbs
+    │   │   └── loader.hbs
+    │   ├── guides
+    │   │   ├── 5-minute-quickstart.hbs
+    │   │   ├── index.hbs
+    │   │   ├── kong-architecture-overview.hbs
+    │   │   ├── kong-ee-introduction.hbs
+    │   │   ├── kong-vitals.hbs
+    │   │   └── uploading-spec.hbs
+    │   ├── index.hbs
+    │   ├── unauthenticated
+    │   │   ├── 404.hbs
+    │   │   ├── index.hbs
+    │   │   ├── login.hbs
+    │   │   ├── register.hbs
+    │   │   └── unauthorized.hbs
+    │   └── user.hbs
+    ├── partials
+    │   ├── custom-js.hbs
+    │   ├── footer.hbs
+    │   ├── header.hbs
+    │   ├── kong-docs-sidebar.hbs
+    │   ├── layout.hbs
+    │   ├── sidebar-spec.hbs
+    │   ├── sidebar.hbs
+    │   ├── spec-dropdown.hbs
+    │   ├── spec-renderer.hbs
+    │   └── unauthenticated
+    │       ├── auth-js.hbs
+    │       ├── code-snippet-languages.hbs
+    │       ├── custom-css.hbs
+    │       ├── footer.hbs
+    │       ├── header.hbs
+    │       ├── layout.hbs
+    │       ├── login-actions.hbs
+    │       └── title.hbs
+    └── specs
+        ├── admin.json
+        ├── files.yaml
+        └── vitals.yaml
+```
+
+When the file structure is setup, you can make use of [GitHub Webhooks](https://developer.github.com/webhooks/creating/) and every time you merge a change into the Master branch, it triggers a push of files to the portal. This allows your developers to work in an environment they are familiar with, and maintain a history of changes to your portal. It also means you can easily roll back to a previous version.
```

Comment:
While this is great advice, I think it is too incomplete to be useful - Github Webhooks isn't enough - there needs to be a process running somewhere that the webhook triggers. Either an example of that process should be provided, or this section should be clarified to explain that.

Reply:
The people who use our product know how to build a trigger for a webhook. Users will have different systems to account for in this process and giving an example will make people want to use that approach or the tools shown which is not the intention.